sound/soc/sof/ipc4.c

```diff
@@ -369,6 +369,17 @@ static int sof_ipc4_tx_msg(struct snd_sof_dev *sdev, void *msg_data, size_t msg_
 	if (!msg_data)
 		return -EINVAL;
 
+	if (!no_pm) {
+		const struct sof_dsp_power_state target_state = {
+			.state = SOF_DSP_PM_D0,
+		};
+
+		/* ensure the DSP is in D0 before sending a new IPC */
```

Comment:
the comment is misleading. D0i3 is a sub-state of D0. So what you meant here is the DSP is in D0i0 - or NOT D0i3

Reply:
let's use d0i0